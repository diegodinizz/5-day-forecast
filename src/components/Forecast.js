import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  width: 90%;
  height: 30px;
  padding-top: 8px;
  font-size: 18px;
`

const DayContainer = styled.span`
  width: 45%;
`

const ForecastIconContainer = styled.div`
  width: 33%;

  img {
    width: 30px;
    height: 30px;
  }
`

const MinMaxContainer = styled.div`
  width: 22%;
  display: flex;
  justify-content: space-between;

  span {
    font-weight: 600;

    :last-child {
      font-weight: 400;
    }
  }
`

export const Forecast = ({ date, forecastIcon, tempMax, tempMin }) => {
  return (
    <Container>
      <DayContainer>{date}</DayContainer>
      <ForecastIconContainer>
        <img
          src={`https://openweathermap.org/img/wn/${forecastIcon}@2x.png`}
          alt='forecast-icon'
        />
      </ForecastIconContainer>
      <MinMaxContainer>
        <span>{tempMax}&deg;</span>
        <span>{tempMin}&deg;</span>
      </MinMaxContainer>
    </Container>
  )
}
