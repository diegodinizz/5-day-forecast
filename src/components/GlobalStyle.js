import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
body {
  font-family: 'Open Sans Condensed';
  letter-spacing: 1px;
  /* padding: 20px 60px; */
  background: -webkit-gradient(linear, 77% 18%, 10% 21%, from(#b9d7ea), to(#769fcd));

  @media(max-width: 800px) {
    padding: 30px;
  }
}

h1 {
  text-align: center;
  margin: 60px 0px 30px 0px;
  font-size: 40px;
  color: #41444b;

  @media(max-width: 800px) {
    margin-top: 0px;
  }
}
`
