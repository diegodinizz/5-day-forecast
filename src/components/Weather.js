import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 20vw;
  height: 29em;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  border: 1px solid #52575d;
  border-radius: 5px;
  background-color: rgba(18, 28, 31, 0.3);
  color: whitesmoke;
  -webkit-box-shadow: 5px 5px 5px 0px rgba(82, 87, 93, 0.5);
  -moz-box-shadow: 5px 5px 5px 0px rgba(82, 87, 93, 0.5);
  box-shadow: 5px 5px 5px 0px rgba(82, 87, 93, 0.5);

  @media (max-width: 800px) {
    width: 100%;
  }

  img {
    width: 48px;
    height: 48px;
  }
`

const CityContainer = styled.div`
  font-size: 2em;
  font-weight: 600;
`

const DescriptionContainer = styled.div`
  font-size: 1.3em;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-right: 20px;
`

const TemperatureContainer = styled.div`
  font-size: 6.5em;
  margin: 10px 0px 20px 20px;
`

const DetailsContainer = styled.div`
  width: 80%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  font-size: 1em;

  span {
    margin-top: 10px;
    font-weight: 600;

    :last-child {
      justify-self: end;
    }
  }
`

const MinMaxContainer = styled(DetailsContainer)`
  border-bottom: 1px solid whitesmoke;

  span {
    margin-bottom: 15px;
    font-weight: 500;
  }
`

export const Weather = ({
  name,
  country,
  temp,
  description,
  tempMin,
  tempMax,
  sunset,
  sunrise,
  feelsLike,
  humidity,
  weatherIcon
}) => {
  return (
    <Container>
      <CityContainer>
        {name}, {country}
      </CityContainer>
      <DescriptionContainer>
        <img
          src={`https://openweathermap.org/img/wn/${weatherIcon}@2x.png`}
          alt='weather-icon'
        />
        {description}
      </DescriptionContainer>
      <TemperatureContainer>{temp}&deg;</TemperatureContainer>
      <MinMaxContainer>
        <span>Min: {tempMin}&deg;</span>
        <span>Max: {tempMax}&deg;</span>
      </MinMaxContainer>
      <DetailsContainer>
        <span>Sunrise: {sunrise}</span>
        <span>Sunset: {sunset}</span>
      </DetailsContainer>
      <DetailsContainer>
        <span>Feels Like: {feelsLike}&deg;</span>
        <span>Humidity: {humidity}%</span>
      </DetailsContainer>
    </Container>
  )
}
