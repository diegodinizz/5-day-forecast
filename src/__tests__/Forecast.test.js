import React from 'react'
import { mount, shallow } from 'enzyme';
import { Forecast } from '../components/Forecast';
import toJSON from 'enzyme-to-json';

const fakeForecast =  {
    date: "Monday",
    forecastIcon: "01n",
    tempMax: 12,
    tempMin: 7,
};

describe('<Forecast/>', () => {
    it('renders and matches the snapshot', () => {
      const wrapper = mount(<Forecast {...fakeForecast} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders the correct day', () => {
        const wrapper = shallow(<Forecast {...fakeForecast} />);
        expect(wrapper.text().includes(fakeForecast.date)).toBe(true);
    });

    it('renders the correct icon properly', () => {
        const wrapper = shallow(<Forecast {...fakeForecast} />);
        const img = wrapper.find('img');
        expect(img.props().src).toContain(fakeForecast.forecastIcon);
    });
});
  