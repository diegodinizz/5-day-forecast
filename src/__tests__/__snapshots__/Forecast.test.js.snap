// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast/> renders and matches the snapshot 1`] = `
<Forecast
  date="Monday"
  forecastIcon="01n"
  tempMax={12}
  tempMin={7}
>
  <styled.div>
    <div
      className="sc-AxjAm gkGyEG"
    >
      <styled.span>
        <span
          className="sc-AxirZ eOkyMB"
        >
          Monday
        </span>
      </styled.span>
      <styled.div>
        <div
          className="sc-AxiKw gBvfsy"
        >
          <img
            alt="forecast-icon"
            src="https://openweathermap.org/img/wn/01n@2x.png"
          />
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-AxhCb eOcLQV"
        >
          <span>
            12
            °
          </span>
          <span>
            7
            °
          </span>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Forecast>
`;
